bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/FileFormat.java

```diff
@@ -84,6 +87,16 @@ List<ElementAtZoom<ImageData>> loadFromByteStream(int fileZoom, int targetZoom)
 		}
 	}
 
+	static abstract class DynamicImageFileFormat extends FileFormat {
```

Comment:
Since the FileFormat itself it already rather dynamic since the latest refactoring and since there is only a single implementation of this class far (and no others are planned to be added), we could remove this intermediate class for now and only keep it for the static format, for which different implementations exist. So as a result, the `SVGFileFormat` could directly implement `FileFormat`.